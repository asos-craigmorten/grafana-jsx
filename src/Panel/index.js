import PropTypes from "prop-types";

const Panel = ({
  type,
  title,
  x,
  y,
  width: w = 12,
  height: h = 9,
  options,
  ...props
}) => {
  const gridPos = {
    x,
    y,
    w,
    h,
  };

  return (
    <panel
      gridPos={gridPos}
      options={options}
      title={title}
      type={type}
      {...props}
    />
  );
};

Panel.PropTypes = {
  title: PropTypes.string,
  type: PropTypes.string.isRequired,
  x: PropTypes.number.isRequired,
  y: PropTypes.number.isRequired,
  width: PropTypes.number,
  height: PropTypes.number,
  options: PropTypes.object,
};

export default Panel;
