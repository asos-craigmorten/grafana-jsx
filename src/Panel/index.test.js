import Panel from "./";

const requiredProps = {
  type: "test-type",
  x: 0,
  y: 0,
};

const withAllProps = {
  ...requiredProps,
  width: 24,
  height: 18,
};

describe("Panel", () => {
  describe("when passed required props", () => {
    it("should render a default panel with required props", () => {
      expect(<Panel {...requiredProps} />).toMatchSnapshot();
    });
  });

  describe("when passed all props", () => {
    it("should render a custom panel", () => {
      expect(<Panel {...withAllProps} />).toMatchSnapshot();
    });
  });

  describe("when passed custom props", () => {
    it("should render a custom panel with the custom props", () => {
      expect(
        <Panel {...withAllProps} test-panel-prop={Symbol("test-panel-value")} />
      ).toMatchSnapshot();
    });
  });
});
