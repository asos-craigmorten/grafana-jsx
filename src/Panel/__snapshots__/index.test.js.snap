// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel when passed all props should render a custom panel 1`] = `
Object {
  "gridPos": Object {
    "h": 18,
    "w": 24,
    "x": 0,
    "y": 0,
  },
  "type": "test-type",
}
`;

exports[`Panel when passed custom props should render a custom panel with the custom props 1`] = `
Object {
  "gridPos": Object {
    "h": 18,
    "w": 24,
    "x": 0,
    "y": 0,
  },
  "test-panel-prop": Symbol(test-panel-value),
  "type": "test-type",
}
`;

exports[`Panel when passed required props should render a default panel with required props 1`] = `
Object {
  "gridPos": Object {
    "h": 9,
    "w": 12,
    "x": 0,
    "y": 0,
  },
  "type": "test-type",
}
`;
